src/CMakeLists.txt

```diff
@@ -17,6 +17,19 @@ target_include_directories(micm
     $<INSTALL_INTERFACE:include>
 )
 
+# Check for Intel compiler
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Intel" OR CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM")
+  # Set compiler flags for Intel C++
+  if (WIN32)
+    # For Windows
+    target_compile_options(micm INTERFACE "/fp:precise")
+  else ()
+    # For other operating systems (e.g., Linux, macOS)
+    target_compile_options(micm INTERFACE "-ffp-model=precise")
+  endif ()
+endif ()
```

Comment:
@sjsprecious I added these compiler flags and I was able to go back to exact equality

Reply:
I tried the `double` option, but icpx didn't recognize that as a keyword. However, `precise` worked